Fix pwm clock
Add pwm pin configuration
Add pwm ressource

Signed-off-by: Nicolas Colombain <nicolas.colombain@armadeus.com>
Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: linux-2.6.27.13/arch/arm/plat-mxc/include/mach/iomux-mx1-mx2.h
===================================================================
--- linux-2.6.27.13.orig/arch/arm/plat-mxc/include/mach/iomux-mx1-mx2.h	2009-04-12 16:10:13.000000000 +0200
+++ linux-2.6.27.13/arch/arm/plat-mxc/include/mach/iomux-mx1-mx2.h	2009-04-12 16:05:57.000000000 +0200
@@ -372,8 +372,6 @@ extern void mxc_gpio_release_multiple(co
 #define PD23_PF_CSPI2_MISO (GPIO_PORTD | GPIO_IN | GPIO_PF  | 23)
 #define PD24_PF_CSPI2_MOSI (GPIO_PORTD | GPIO_OUT | GPIO_PF  | 24)
 
-
-
 #define PD25_PF_CSPI1_RDY	(GPIO_PORTD | GPIO_OUT | GPIO_PF  | 25)
 #define PD26_PF_CSPI1_SS2	(GPIO_PORTD | GPIO_OUT | GPIO_PF  | 26)
 #define PD26_AF_USBH2_DATA5     (GPIO_PORTD | GPIO_AF | 26)
@@ -388,6 +386,7 @@ extern void mxc_gpio_release_multiple(co
 #define PE2_PF_USBOTG_DIR	(GPIO_PORTE | GPIO_OUT | GPIO_PF | 2)
 #define PE3_PF_UART2_CTS	(GPIO_PORTE | GPIO_OUT | GPIO_PF | 3)
 #define PE4_PF_UART2_RTS	(GPIO_PORTE | GPIO_IN  | GPIO_PF | 4)
+#define PE5_PF_PWM0			(GPIO_PORTE | GPIO_OUT | GPIO_PF | 5)
 #define PE6_PF_UART2_TXD	(GPIO_PORTE | GPIO_OUT | GPIO_PF | 6)
 #define PE7_PF_UART2_RXD	(GPIO_PORTE | GPIO_IN  | GPIO_PF | 7)
 #define PE8_PF_UART3_TXD	(GPIO_PORTE | GPIO_OUT | GPIO_PF | 8)
Index: linux-2.6.27.13/arch/arm/mach-mx2/devices.c
===================================================================
--- linux-2.6.27.13.orig/arch/arm/mach-mx2/devices.c 2009-04-12 16:10:13.000000000 +0200
+++ linux-2.6.27.13/arch/arm/mach-mx2/devices.c	2009-04-12 16:04:25.000000000 +0200
@@ -637,6 +637,25 @@ struct platform_device mxc_ehci2 = {
 	.resource = mxc_ehci2_resources,
 };
 
+static struct resource pwm_resources[] = {
+    [0] = {
+        .start  = (PWM_BASE_ADDR),
+        .end    = (PWM_BASE_ADDR + 0x14),
+        .flags  = IORESOURCE_MEM,
+    },
+    [1] = {
+        .start  = (MXC_INT_PWM),
+        .end    = (MXC_INT_PWM),
+        .flags  = IORESOURCE_IRQ, 
+    },
+};
+
+struct platform_device imx_pwm_device0 = {
+    .name           = "imx-pwm", // Should be the same as the driver name
+    .id             = 0,
+    .num_resources  = ARRAY_SIZE(pwm_resources),
+    .resource       = pwm_resources,
+};
 
 /* GPIO port description */
 static struct mxc_gpio_port imx_gpio_ports[] = {
Index: linux-2.6.27.13/arch/arm/mach-mx2/devices.h
===================================================================
--- linux-2.6.27.13.orig/arch/arm/mach-mx2/devices.h 2009-04-12 16:10:13.000000000 +0200
+++ linux-2.6.27.13/arch/arm/mach-mx2/devices.h	2009-04-12 16:05:02.000000000 +0200
@@ -29,3 +29,4 @@ extern struct platform_device mxc_dam_de
 extern struct platform_device mxc_otg;
 extern struct platform_device mxc_ehci1;
 extern struct platform_device mxc_ehci2;
+extern struct platform_device imx_pwm_device0;
Index: linux-2.6.27.13/arch/arm/mach-mx2/clock_imx27.c
===================================================================
--- linux-2.6.27.13.orig/arch/arm/mach-mx2/clock_imx27.c 2009-04-12 16:10:12.000000000 +0200
+++ linux-2.6.27.13/arch/arm/mach-mx2/clock_imx27.c	2009-04-12 16:06:29.000000000 +0200
@@ -921,7 +921,7 @@ static struct clk pwm_clk[] = {
 		.parent = &per_clk[0],
 		.secondary = &pwm_clk[1],
 	}, {
-		.name = "pwm_clk",
+		.name = "pwm_ipg_clk",
 		.parent = &ipg_clk,
 		.enable = _clk_enable,
 		.enable_reg = CCM_PCCR0,


